streams/src/main/java/org/apache/kafka/streams/kstream/Materialized.java

```diff
@@ -103,6 +110,21 @@ protected Materialized(final Materialized<K, V, S> materialized) {
         return new Materialized<>(storeName);
     }
 
+    /**
+     * Materialize a {@link StateStore} with the store implementation.
+     *
+     * @param storeImplementation  store implementation used to materialize the store
+     * alphanumerics, '.', '_' and '-'.
+     * @param <K>       key type of the store
+     * @param <V>       value type of the store
+     * @param <S>       type of the {@link StateStore}
+     * @return a new {@link Materialized} instance with the given storeName
+     */
+    public static <K, V, S extends StateStore> Materialized<K, V, S> as(final StoreImplementation storeImplementation) {
```

Comment:
new overloading method for `Materialized#as`

Reply:
In practice I thought no users code would call this function since they are all set via the config? I have not seen any non-testing code that calls this.

Reply:
The reason I added this method is because user can still override the default config via this method if they want.